system/security/securesocket/securesocket.cpp

```diff
@@ -1813,54 +1812,23 @@ class CRsaCertificate : implements ICertificate, public CInterface
 } // namespace securesocket
 
 extern "C" {
-CriticalSection factoryCrit;
 
 SECURESOCKET_API ISecureSocketContext* createSecureSocketContext(SecureSocketType sockettype)
 {
-    CriticalBlock b(factoryCrit);
-    if(sockettype == ClientSocket)
-    {
-        return new securesocket::CSecureSocketContext(sockettype);
-    }
-    else
-    {
-        if(server_securesocket_context.get() == NULL)
-            server_securesocket_context.setown(new securesocket::CSecureSocketContext(sockettype));
-        return server_securesocket_context.getLink();
-    }
+    return new securesocket::CSecureSocketContext(sockettype);
```

Comment:
I think we decided to try and use the same context.  Why this change ?

Reply:
ok, answered

Reply:
yes - description in commit comment